client/landing/stepper/declarative-flow/internals/steps-repository/design-choices/index.tsx

```diff
@@ -105,15 +98,12 @@ const DesignChoicesStep: Step = ( { navigation, flow, stepName } ) => {
 											}
 										)
 									) }
-									onSelect={ () => {
+									onSelect={ ( destination ) => {
 										recordTracksEvent( 'calypso_big_sky_choose', {
 											flow,
 											step: stepName,
 										} );
-										const queryParams = new URLSearchParams( location.search ).toString();
-										navigate(
-											`/setup/site-setup/launch-big-sky${ queryParams ? `?${ queryParams }` : '' }`
-										);
+										handleSubmit( destination );
 									} }
```

Comment:
I had to make changes to the existing use of the design choices step.

It's accessible when you go through onboarding _without_ goals first. If you choose a premium or higher plan then you're shown the Big Sky choice in the `site-setup` flow.

This logic for navigating to Big Sky was added directly to the step component in this diff: https://github.com/Automattic/wp-calypso/commit/ba9641060527779cb7200e99680e06816d975eb5#diff-a9c1ce51ac0fbf560d78a2bbbb650b6edd25cb528c840a1e435dfb752c2eae4fR131

The flow navigation logic needs to live inside `site-setup-flow.ts` so we can customise the logic depending on what flow the step is included in.

Reply:
Thanks for following this good stepper practice. 
Here's a great follow-up doc if someone wants to learn more about why this is important: pdDR7T-1M7-p2